packages/liveblocks-react/src/room.tsx

```diff
@@ -1400,6 +1400,20 @@ function useDeleteThread(): (threadId: string) => void {
       const optimisticUpdateId = nanoid();
 
       const { store, onMutationFailure } = getExtrasForClient(client);
+
+      const thread = store.get().threads[threadId];
+
+      const userId = getCurrentUserId(room);
+
+      if (
+        thread &&
+        thread.comments &&
+        thread.comments[0] &&
+        thread.comments[0].userId !== userId
+      ) {
+        throw new Error("Only the thread creator can delete the thread");
```

Comment:
I am trying to recall if the public API endpoint also returns deleted comments to ensure that the first comment under a thread would always come from the thread owner.

Reply:
I think it isn’t if user A creates the thread, then user B leaves a comment, and user A delete their initial thread‘s comment. Now user B’s comment will be the new thread‘s “root”, so I guess it makes sense to declare user B the new thread owner, right?

Reply:
Owned by inheritance.

Reply:
As long as the backend is also consistent with what we do in the frontend. Otherwise, we will be preventing thread deletion too eagerly.

Reply:
It looks like deleted comments are returned in the API response. So, in your example @nvie, User A is still the owner of the thread. Backend also seems to think the same.

Reply:
@FlowFlorent We also need to make sure when a comment is deleted, the optimistic update only soft deletes the comment just to be sure? The thread  should, nonetheless be replaced very quickly with the most recent value after receiving a websocket event.

Reply:
Agreed, being consistent with the backend is a must.

Reply:
We soft delete comments through the client API so the owner of the thread will always be the original thread creator.

There's one area where this changes, the REST API endpoint to delete a comment hard deletes the comment, meaning that the owner could change by inheritance. Wonder if we should set it as a soft delete instead 🤔

Reply:
The optimistic update indeed soft deletes the comment, so we should be good! Thanks for raising the point, I forgot to check that!
```ts
  const updatedComments = thread.comments.map((comment) =>
    comment.id === commentId
      ? {
          ...comment,
          deletedAt,
          body: undefined,
        }
      : comment
  );
```